### ISSUE TITLE
Name conflict with Mongoose embedded web server

### ISSUE THREAD

**Issue by mmuetzel** [DrTimothyAldenDavis/SuiteSparse#587]
The library libmongoose conflicts with the library of the same name from the Mongoose embedded web server project:
https://github.com/cesanta/mongoose

The oldest commit in that repository is from 2010. So, that conflict is probably already there for a while.

This can cause issues if projects would like to use both libraries from SuiteSparse and from that embedded web server at the same time.

Prompted by a recent report for the MSYS2 project:
https://github.com/msys2/MINGW-packages/issues/19344

**Comment by DrTimothyAldenDavis** [comment]
Ok. I could rename it, given that their name came first. Or nearly so.  We first came out with Mongoose around the same time. It just wasn't on github then. 

Say instead of libmongoose.so (DLL etc) it could be libsuitesparse_mongoose.so. Where things are capitalized, 
Like the cmake files, would be SuiteSparse_Mongoose* instead. 

I suppose the cmake namespace could be SuiteSparse::Mongoose still?  

That's a bit wordy but at least it's abundantly clear what the new name refers to. 

I'm tied up the next few days and will be away from the keyboard most of the time. This won't be a big change though. 

Does this sound like a good plan?

**Comment by DrTimothyAldenDavis** [comment]
A few moments this afternoon at the keyboard.   I also noticed this with homebrew, which has a copy of the web server mongoose.

 It looks like the web-server namespace is `mg_*` and it's a C library, not C++.  Its include file is mongoose.h.  They don't have a mongooseConfig.cmake.

So that doesn't conflict with SuiteSparse/Mongoose, which is C++ with all its functions named `Mongoose*` and the header file is `Mongoose.hpp`.

It seems like the only conflict is the name of the compiled library, libmongoose.so (dll, dylib, a, etc). 

I think the least disruptive thing would be to keep the SuiteSparse/Mongoose source code and #include file (Mongoose.hpp) the same.  I should rename my `libmongoose.*` to `libsuitesparse_mongoose.*.`  I should probably rename all the Mongoose cmake and pkg-config files from `MongooseConfig.cmake` (etc) to `SuiteSparse_MongooseConfig.cmake` (etc) so there's no confusion with a `find_package ( SuiteSparse_Mongoose ...)` in an end-user cmake file, in case the web server mongoose project ever creates their own mongooseConfig.cmake.  Some OS's  are not case sensitive and that would break.

I'll start making this change.  Let me know if think I need to make more changes.

**Comment by DrTimothyAldenDavis** [comment]
See https://github.com/DrTimothyAldenDavis/SuiteSparse/pull/588

**Comment by DrTimothyAldenDavis** [comment]
It wasn't too hard to make this change, and the changes are pretty minimal.  The cmake build script of the Example package had to change just a bit.  I don't think this will be too disruptive to any application using SuiteSparse.  Their code will remain the same, just link with -lsuitesparse_mongoose instead of -lmongoose.

**Comment by DrTimothyAldenDavis** [comment]
This is now fixed in 7.4.0.beta6 (to be released soon).
